### PR TITLE
[Azure Logs] Clarify the Storage Account container role in the integration

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

- Extend the Azure Logs documentation on the Storage Account. We want to make sure users understand why it is needed and how to set it up to deploy this integration successfully.
-  Add a small new "Requirements and setup" section in each integration page that links to the general Azure Logs page, containing the requirements and detailed setup instructions.

### Motivation

We received some questions about the role of the Storage Account in the Azure Logs integration. At least one user—probably more—couldn't find the needed information on our public-facing documentation. 

Here are the questions:

```text
- Do we need to set up and Azure storage account for the integration to work?

- Is it enough with just having an event hub?

- Why is recommended setting up an storage account and how does it help?

- What can happen if we stop using the storage account with the integration?

- If we need to use the storage account, what configuration elastic recommends? Hot or cold storage, how long to store the data in the storage, etc
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Screenshots

Here is a screenshot of the link to the general Azure Logs page that contains requirements and setup information.

CleanShot 2023-03-22 at 20 27 47@2x

### CODE REVIEW

**Comment by zmoog** [comment]
You will see the changes twice, but this is normal. The integration stores the doc in source and rendered forms.

**Review by endorama — APPROVED** [review]

**Comment by alstolten** [comment]
Hey @zmoog LGTM. I believe we can take it from here and adjust for more user feedback once collected. Thanks for the changes!

**Review by alstolten — APPROVED** [review]

File: packages/azure/_dev/build/docs/eventhub.md [hunk 1]
```diff
@@ -1,11 +1,19 @@
 # Azure Eventhub Input
 
 The Azure Eventhub Input integration allows users to collect events from Azure event hubs.
- The azure-eventhub input functionality is based on the the event processor host (EPH is intended to be run across multiple processes and machines while load balancing message consumers more on this here https://github.com/Azure/azure-event-hubs-go#event-processor-host, https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host). State such as leases on partitions and checkpoints in the event stream are shared between receivers using an Azure Storage container. 
+
+The azure-eventhub input functionality is based on the the event processor host (EPH is intended to be run across multiple processes and machines while load balancing message consumers more on this here https://github.com/Azure/azure-event-hubs-go#event-processor-host, https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host). State such as leases on partitions and checkpoints in the event stream are shared between receivers using an Azure Storage container.
```
Comment (dmathieu) [inline packages/azure/_dev/build/docs/eventhub.md:5]: I know this was already like this before, but isn't it a bit strange to have URLs in a doc? Wouldn't they be more readable as links? Something like:

```suggestion
The azure-eventhub input functionality is based on the the event processor host (EPH is intended to be run across multiple processes and machines while load balancing message consumers more on this in the [Azure event-hubs-go doc](https://github.com/Azure/azure-event-hubs-go#event-processor-host) and [Azure event-processor doc](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host)). State such as leases on partitions and checkpoints in the event stream are shared between receivers using an Azure Storage container. 
```

File: packages/azure/_dev/build/docs/eventhub.md [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+
+The azure-eventhub input functionality is based on the the event processor host (EPH is intended to be run across multiple processes and machines while load balancing message consumers more on this here https://github.com/Azure/azure-event-hubs-go#event-processor-host, https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host). State such as leases on partitions and checkpoints in the event stream are shared between receivers using an Azure Storage container. 
  For this reason, as a prerequisite to using this input, users will have to create or use an existing storage account.
 
 There are several requirements before using the integration since the logs will actually be read from azure event hubs.
-   * the logs/metrics have to be exported first to the event hub https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create-kafka-enabled
+
+* the logs/metrics have to be exported first to the event hub https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create-kafka-enabled
+
+There are several requirements before using the integration since the logs will actually be read from Azure Event Hubs.
```
Comment (dmathieu) [inline packages/azure/_dev/build/docs/eventhub.md:12]: There is the exact same phrase two lines above. Is that expected?
Also, this says "there are several requirements", but there seems to be only one mentioned (before this line)?

File: packages/azure/_dev/build/docs/springcloudlogs.md [hunk 3]
```diff
@@ -2,13 +2,13 @@
 
 The Azure Logs integration retrieves different types of log data from Azure.
 
-There are several requirements before using the integration since the logs will actually be read from azure event hubs.
+Users opting for Elastic Cloud native Azure integration can stream the Azure Spring Cloud logs directly to their partner solution clusters, more information and steps can be found {{ url "observability-monitor-azure" "here" }}.
 
-- The logs have to be [exported first to the event hub](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create-kafka-enabled).
-- To export activity logs to event hubs users can follow the steps [here](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/activity-log-export).
-- To export audit and sign-in logs to event hubs users can follow the steps [here](https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/tutorial-azure-monitor-stream-logs-to-event-hub).
+There are several requirements before using the integration since the logs will actually be read from Azure Event Hubs.
```
Comment (dmathieu) [inline packages/azure/_dev/build/docs/springcloudlogs.md:7]: Same here, this is a bit weird. You mention several requirements, but then don't say what they are.

File: packages/azure/_dev/build/docs/eventhub.md (hunk 2 above)
Comment (zmoog) [reply packages/azure/_dev/build/docs/eventhub.md:12]: Nope! 🤦

File: packages/azure/_dev/build/docs/eventhub.md (hunk 1 above)
Comment (zmoog) [reply packages/azure/_dev/build/docs/eventhub.md:5]: Yep, I agree. I considered making this change every time I revised these documents for other reasons. Today is the day; thank you for bringing this up in this review! ❤️

File: packages/azure/_dev/build/docs/eventhub.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 The Azure Eventhub Input integration allows users to collect events from Azure event hubs.
 
-The azure-eventhub input functionality is based on the the event processor host (EPH is intended to be run across multiple processes and machines while load balancing message consumers more on this here https://github.com/Azure/azure-event-hubs-go#event-processor-host, https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host). State such as leases on partitions and checkpoints in the event stream are shared between receivers using an Azure Storage container. 
+The azure-eventhub input functionality is based on the the event processor host (EPH is intended to be run across multiple processes and machines while load balancing message consumers more on this in the [Azure event-hubs-go doc](https://github.com/Azure/azure-event-hubs-go#event-processor-host) and [Azure event-processor doc](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-event-processor-host)). State such as leases on partitions and checkpoints in the event stream are shared between receivers using an Azure Storage container.
```
Comment (alstolten) [inline packages/azure/_dev/build/docs/eventhub.md:5]: Two minor suggestions:
`based on the the event` -> remove one "the".
`message consumers more on this in the` -> `message consumers, see`

**Comment by zmoog** [comment]
@dmathieu I took the opportunity of this review to make additional changes to the Azure Logs integration docs.

Now each integration page is aligned with the following broad structure:

1. description and purpose
2. additional info (depending on what is available right now)
3. requirements and setup

Here are a couple of examples:

CleanShot 2023-03-23 at 12 39 12@2x

CleanShot 2023-03-23 at 12 38 34@2x

File: packages/azure/_dev/build/docs/eventhub.md (hunk 4 above)
Comment (zmoog) [reply packages/azure/_dev/build/docs/eventhub.md]: I've added it to the latest commit; please let me know if the update version works for you.

File: packages/azure/docs/springcloudlogs.md [hunk 5]
```diff
@@ -2,13 +2,13 @@
 
 The Azure Logs integration retrieves different types of log data from Azure.
 
-There are several requirements before using the integration since the logs will actually be read from azure event hubs.
+Users opting for Elastic Cloud native Azure integration can stream the Azure Spring Cloud logs directly to their partner solution clusters, more information and steps can be found [here](https://www.elastic.co/guide/en/observability/current/monitor-azure.html).
 
-- The logs have to be [exported first to the event hub](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create-kafka-enabled).
-- To export activity logs to event hubs users can follow the steps [here](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/activity-log-export).
-- To export audit and sign-in logs to event hubs users can follow the steps [here](https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/tutorial-azure-monitor-stream-logs-to-event-hub).
+There are several requirements before using the integration since the logs will actually be read from Azure Event Hubs.
```
[the remark this answers is not in the document]
Comment (zmoog) [reply packages/azure/docs/springcloudlogs.md]: This sentence does not provide much value, so I removed it in the latest update. The presence of the "Requirements and setup" section is probably enough. WDYT?

**Comment by zmoog** [comment]
This documentation requires revision, but the scope is broader than this PR. I am addressing the most critical changes here and saving the rest for another PR.

File: packages/azure/_dev/build/docs/README.md [hunk 6]
```diff
@@ -69,7 +67,7 @@ To learn more about Event Hubs, refer to [Features and terminology in Azure Even
 
 The [Storage account](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview) is a versatile Azure service that allows you to store data in various storage types, including blobs, file shares, queues, tables, and disks.
 
-The Azure Logs integration uses a Storage account container to store and share information about the Consumer Group (state, position, or offset). Sharing such information allows multiple Elastic Agents assigned to the same agent policy to work together; this enables horizontal scaling of the logs processing when required.
+The Azure Logs integration requries a Storage account container to work. The integration uses the Storage account container for checkpointing; it stores data about the Consumer Group (state, position, or offset) and share it among the Elastic Agents. Sharing such information allows multiple Elastic Agents assigned to the same agent policy to work together; this enables horizontal scaling of the logs processing when required.
```
Comment (dmathieu) [inline packages/azure/_dev/build/docs/README.md:70]:
```suggestion
The Azure Logs integration requieres a Storage account container to work. The integration uses the Storage account container for checkpointing; it stores data about the Consumer Group (state, position, or offset) and shares it among the Elastic Agents. Sharing such information allows multiple Elastic Agents assigned to the same agent policy to work together; this enables horizontal scaling of the logs processing when required.
```

File: packages/azure/docs/README.md [hunk 7]
```diff
@@ -69,7 +67,7 @@ To learn more about Event Hubs, refer to [Features and terminology in Azure Even
 
 The [Storage account](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview) is a versatile Azure service that allows you to store data in various storage types, including blobs, file shares, queues, tables, and disks.
 
-The Azure Logs integration uses a Storage account container to store and share information about the Consumer Group (state, position, or offset). Sharing such information allows multiple Elastic Agents assigned to the same agent policy to work together; this enables horizontal scaling of the logs processing when required.
+The Azure Logs integration requries a Storage account container to work. The integration uses the Storage account container for checkpointing; it stores data about the Consumer Group (state, position, or offset) and share it among the Elastic Agents. Sharing such information allows multiple Elastic Agents assigned to the same agent policy to work together; this enables horizontal scaling of the logs processing when required.
```
[the remark this answers is not in the document]
Comment (dmathieu) [reply packages/azure/docs/README.md:70]: I'm not seeing these typos being resolved in the latest diff.

**Comment by zmoog** [comment]
@dmathieu, do you think this needs more work or we're good to merge?

**Review by dmathieu — APPROVED** [review]
:shipit:

**Pull request merged**